### DISCUSSION TITLE
Failing to automate slide creation for each plot (pptx format, R)

### DISCUSSION THREAD

**Discussion by Yann-C-INN** [quarto-dev/quarto-cli#10348] [Q&A] [answered]
I have a list with 100 plots which I want to automate via quarto such that each plot is produced in a new slide of a power point presentation.
I am using `Quarto 1.4.555` with `R 4.4.0` via `RStudio 2024.4.2.764`

Here is a reproducible example of what I did:
````qmd
---
title: "test-auto"
format: pptx
---

```{r setup, include=FALSE}
library(knitr)
library(tidyverse)
```

```{r load-data}
plot_list_q <-  list(
  ggplot(mtcars, aes(x = mpg, y = wt)) + geom_point(),
  ggplot(mtcars, aes(x = hp, y = drat)) + geom_point()
)

```

```{r function}
# Slide creation function
create_slide <- function(plot, index) {
  knit_expand(text = c(
    paste("## Plot", index),
    "",
    "```{r}",
    "plot_list_q[[{{index}}]]",
    "```"
  ))
}
```

```{r generate}
#| echo: false
#| results: asis

# Generate all slides
slides <- map2(plot_list_q, seq_along(plot_list_q), create_slide)
cat(unlist(slides), sep = "\n")
```

````

There are indeed two plot slides, but the plot is not generated:
[![enter image description here](https://i.sstatic.net/7AGwRRoe.png)](https://i.sstatic.net/7AGwRRoe.png)

Would love to get some community insight. I hope to avoid manually creating these slides...

**Comment by Yann-C-INN** [comment, marked as answer]
Julian from SO:
```
create_slide <- function(plot, index) {
    knitr::knit_child(text = c(
      paste("## Plot", index),
      "",
      "```{r}",
      "plot_list_q[[", index, "]]",
      "```"
    ), envir = environment(), quiet = TRUE)
  }
```

**Reply by mcanouil** [reply]
@Yann-C-INN I wanted to kindly remind you that cross-posting the same question or issue in multiple places can sometimes be seen as disrespectful. This is because it can lead to multiple people spending time on the same problem without knowing if it has already been resolved elsewhere.
To ensure that everyone can collaborate effectively and avoid duplicated efforts, it’s best to post your question or issue in one place and wait for a response. This way, we can all work together more efficiently and help you out as quickly as possible.
Thank you for understanding!

**Reply by Yann-C-INN** [reply]
It is a fair point. 
I think that closing my own topic the minute I received an answer elsewhere + providing the solution for others to benefit aligns pretty well with community standards. It's not like i left the topic open and forgot about it...

Thanks for your feedback!
